File: android-wrapper/app/src/main/java/org/m0skit0/android/hms/unity/ads/RewardAdLoadListenerWrapper.java
package org.m0skit0.android.hms.unity.ads;

public class RewardAdLoadListenerWrapper extends com.huawei.hms.ads.reward.RewardAdLoadListener {

    private final RewardAdLoadListener listener;

    public RewardAdLoadListenerWrapper(final RewardAdLoadListener listener) {
        super();
        this.listener = listener;
    }

    @Override
    public void onRewardAdFailedToLoad(int errorCode) {
        super.onRewardAdFailedToLoad(errorCode);
        listener.onRewardAdFailedToLoad(errorCode);
    }

    @Override
    public void onRewardedLoaded() {
        super.onRewardedLoaded();
        listener.onRewardedLoaded();
    }
}

File: android-wrapper/app/src/main/java/org/m0skit0/android/hms/unity/ads/RewardAdLoadListenerWrapper.kt
package org.m0skit0.android.hms.unity.ads

class RewardAdLoadListenerWrapper(private val listener: RewardAdLoadListener) : com.huawei.hms.ads.reward.RewardAdLoadListener() {

    override fun onRewardedLoaded() {
        listener.onRewardedLoaded()
    }

    override fun onRewardAdFailedToLoad(errorCode: Int) {
        listener.onRewardAdFailedToLoad(errorCode)
    }
}